src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -20,17 +21,46 @@ export default class SimidMapCreative extends BaseSimidCreative {
   }
 
   /** @override */
-  onStart(eventData){
-    super.onStart(eventData);
-    //ToDo(juliareichel@): handle invalid JSON and param errors
-    const adParams = JSON.parse(this.creativeData.adParameters);
+  onInit(eventData) {
+    this.validateAndParseAdParams_(eventData);
+    super.onInit(eventData);
```

Comment:
If there is invalid data, then you'd send a reject message, but then this will still run and you'd send a resolve message anyway, right?

I think that would be a problem.  You probably need to find a way to only send resolve if all these functions pass properly. It might require some refactoring of base_simid_creative or simid_protocol (it might not, there is probably solutions that don't touch those as well)

Reply:
fixed